perfkitbenchmarker/openstack/utils.py

```diff
@@ -0,0 +1,183 @@
+                    'defaults to $OPENSTACK_PASSWORD_FILE. Alternatively, '
+                    'setting the password itself in $OS_PASSWORD is also '
+                    'supported.')
+
+
+class KeystoneAuth(object):
+    """
+        Usage example:
+        auth = KeystoneAuth(auth_url, auth_tenant, auth_user, auth_password)
+        token = auth.get_token()
+        tenant_id = auth.get_tenant_id()
+
+        token and tenant_id are required to use all OpenStack python clients
+    """
+
+    def __init__(self, url, tenant, user, password):
+        self.__url = url
+        self.__tenant = tenant
+        self.__user = user
+        self.__password = password
+        self.__connection = None
+        self.__session = None
+
+    def __getattribute__(self, item):
+        if item == '_KeystoneAuth__connection':
+            if self.__dict__['_KeystoneAuth__connection'] is None:
+                self.__authenticate()
+
+        return super(KeystoneAuth, self).__getattribute__(item)
+
+    def __authenticate(self):
+        import keystoneclient.v2_0.client as ksclient
+
+        self.__connection = ksclient.Client(
+            auth_url=self.__url,
+            username=self.__user,
+            password=self.__password,
+            tenant=self.__tenant)
+        self.__connection.authenticate()
+
+    def get_token(self):
+        return self.__connection.get_token(self.__session)
+
+    def get_tenant_id(self):
+        raw_token = self.__connection.get_raw_token_from_identity_service(
+            auth_url=self.__url,
+            username=self.__user,
+            password=self.__password,
+            tenant_name=self.__tenant
+        )
+        return raw_token['token']['tenant']['id']
+
+
+class NovaClient(object):
+    instance = None
+
+    def __new__(cls, *args):
+        if not NovaClient.instance:
+            cls.instance = object.__new__(cls, *args)
+        return cls.instance
+
+    def __getattribute__(self, item):
+        try:
+            return super(NovaClient, self).__getattribute__(item)
+        except AttributeError:
+            return self.__client.__getattribute__(item)
+
+    def GetPassword(self):
+      # For compatibility with Nova CLI, use 'OS'-prefixed environment value
+      # if present. Also support reading the password from a file.
+
+      error_msg = ('No OpenStack password specified. '
+                   'Either set the environment variable OS_PASSWORD to the '
+                   'admin password, or provide the name of a file '
+                   'containing the password using the OPENSTACK_PASSWORD_FILE '
+                   'environment variable or --openstack_password_file flag.')
+
+      password = os.getenv('OS_PASSWORD')
+      if password is not None:
+        return password
+      try:
+        with open(os.path.expanduser(FLAGS.openstack_password_file)) as pwfile:
+          password = pwfile.readline()
+          password.rstrip()
```

Comment:
rstrip() works on a copy of the object so valid usage would be:
password = password.rstrip() 
return password